engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -1037,4 +1037,59 @@ namespace dmGameSystem
         pit->m_FnIterateNext = CompMeshIterPropertiesGetNext;
     }
 
+    static dmGameObject::Result CompMeshTypeCreate(const dmGameObject::ComponentTypeCreateCtx* ctx, dmGameObject::ComponentType* type)
+    {
+        MeshContext* mesh_context = new MeshContext;
+        mesh_context->m_Factory = ctx->m_Factory;
+        mesh_context->m_RenderContext = *(dmRender::HRenderContext*)ctx->m_Contexts.Get(dmHashString64("render"));
+        mesh_context->m_MaxMeshCount = dmConfigFile::GetInt(ctx->m_Config, "mesh.max_count", 128); // TODO - also initialized in engine.cpp:1106. Why ?
+
+//        int32_t max_gui_count = dmConfigFile::GetInt(ctx->m_Config, "gui.max_instance_count", 128);
+//        gui_context->m_Worlds.SetCapacity(max_gui_count);
+//        dmGui::InitializeScript(gui_context->m_ScriptContext);
+
+        ComponentTypeSetPrio(type, 725);
+
+        ComponentTypeSetContext(type, mesh_context);
+//        ComponentTypeSetHasUserData(type, true);      // TODO  ???
```

Comment:
If you see the original code, it sets it to true.

Reply:
Yeah, i just found out. Also, there is a memset(0) that zeros out all the "type" struct.